Drawsana/Tools/Implementations/SelectionTool.swift

```diff
@@ -27,6 +27,18 @@ public class SelectionTool: DrawingTool {
 
   private var originalTransform: ShapeTransform?
   private var startPoint: CGPoint?
+  /* When you tap away from a shape you've just dragged, the method calls look
+     like this:
+      - handleDragStart (hitTest on selectedShape fails)
+      - handleDragContinue
+      - handleDragCancel
+      - handleTap
+
+     We need to be careful not to incorrectly reset the transform for the selected
+     shape when you tap away, so we explicitly capture whether you are actually
+     dragging the shape or not.
+   */
+  private var isDraggingShape = false
```

Comment:
Note that this was the way I intended tools to be called since the beginning, but the implementation didn't actually do it until now, which revealed this bug.

Every single tap will be preceded by dragStart->dragContinue->dragCancel. Maybe a little confusing, but IMO correct in the context of a drawing program.

Reply:
I think in an ideal world a tap wouldn't be preceded by these events. But it would be somewhat tricky to implement a world in which that happens so let's not do it now.